dashboard/app/controllers/levels_controller.rb

```diff
@@ -295,6 +297,8 @@ def level_params
       params[:level][param].delete_if(&:empty?) if params[:level][param].is_a? Array
     end
 
+    params[:level][:reference_links].delete_if {|link| link == ""}
```

Comment:
Would add a short comment stating why this is necessary (our Ruby code is notoriously under-commented imo, and we should try to be better)